cranelift/codegen/src/isa/mod.rs

```diff
@@ -172,12 +190,12 @@ impl Builder {
     /// flags are inconsistent or incompatible: for example, some
     /// platform-independent features, like general SIMD support, may
     /// need certain ISA extensions to be enabled.
-    pub fn finish(self, shared_flags: settings::Flags) -> CodegenResult<OwnedTargetIsa> {
-        (self.constructor)(self.triple, shared_flags, self.setup)
+    pub fn finish(&self, shared_flags: settings::Flags) -> T {
```

Comment:
Why is this no longer consuming the builder? I don't see a reason why you would want to resue the builder across building multiple TargetIsa. Nornally you did either build once or build multiple times but with different settings.

Reply:
Looking through this a bit, it seems like we don't necessarily have to make this change.

However, you can see [here](https://github.com/bytecodealliance/wasmtime/pull/5944/files#diff-4f5aab5323544d319ac344b7f711167c204d0afb53625530f09e90c190994e0aL103-R72) that previously, even though we were consuming the builder when calling `finish`, we ended up cloning the builder anyways to get around moving the value behind a shared reference. It might make more sense to not explicitly move the value out when calling `finish` if it means the builder is more ergonomic to use in other places. Also cc @alexcrichton in case you have thoughts on this.

Reply:
Ah I've forgotten why I originally did it. At some point along the way when I was originally refactoring I needed to do this to resolve some compile error somewhere but I've lost the context so it might be worth backing out this part and seeing what happens?

Reply:
Yeah I took a stab at moving back! It can be done but it requires cloning the builder in other places (it's `&self` all the way down). I don't have a preference really but I can move it back if that's easier.

Reply:
I gave it a try [here](https://github.com/bytecodealliance/wasmtime/pull/5944/commits/60b6947ffff7e24fc434a008f5f278ead023afb7) and it seems to work fine, but it requires adding clone support in a few more places.

Reply:
Oh sorry forgot to mention just now, but I do not personally have any preference on which way this goes. I don't know of any reason myself to preserve the current functionality nor necessarily any reason to move to the borrowed version.

Reply:
So my commit messed a few things on the `winch` side that's making me think the borrowed version might be a bit easier. Needing to `clone` the builder in a call is a bit tricky, since the winch and Cranelift currently use different return types with the `IsaBuilder`. I could unify them in theory, but I'm thinking it might be better to use the borrowing strategy here while we are still figuring what needs to be common.